ocw/lib/provider.py

```diff
@@ -35,6 +38,31 @@ def get_data(self, name=None) -> str:
             return self.auth_json
         return self.auth_json[name]
 
+    @staticmethod
+    def get_tags(resource) -> dict:
+        try:
+            return resource.tags
+        except AttributeError:
+            return {}
+
+    def is_cancelled(self, resource) -> bool:
+        tags = self.get_tags(resource)
+        if TAG_IGNORE in tags:
+            return False
+        job = tags.get("openqa_var_JOB_ID")
+        if job is None:
+            # Parse "11555642 openqa_created_by=openqa.suse.de/t11555642 openqa_ttl=11100"
+            try:
+                job = tags.get("openqa-cli-test-tag").split()[0]
+            except AttributeError:
+                pass
+        if job is not None:
+            return OpenQA(
+                server=OPENQA_URL,
```

Comment:
Does not this create an hard dependency between the generic PCW code and a setup referring to OSD?

What happens if the job id referred in the `openqa_var_JOB_ID` is not found on OSD instance ? It could happens if an ID does not refer to a job on OSD but on some other instance.

Reply:
... or just means that it is a feature of PCW that only do something for a specific subset of resources:
- resources with a specific tags ...
- ... that refer to a job id that exist on OSD ...
- ... and  is in status cancelled.

The feature remain inactive for anything else. (resource without that tag, resources from job on other openQA instances, resources from job not cancelled but maybe failed).

Reply:
From my POV:
1) Job not found in openQA should be ignored, not deleted.
2) Jobs without `openqa_created_by` and `openqa_ttl` tags should be also ignored.
3) Not only OSD should be supported but also other instances.

Reply:
> Does not this create an hard dependency between the generic PCW code and a setup referring to OSD?

This is already the case within `webui`.  PCW is not generic as this is our tool for cleanup.

The `OpenQA` class introduced here supports multiple servers, though.  It could parse the URL found in `openqa_created_by`.  I know that you're using a dedicated machine for openQA jobs.  That's why I wanted your input.

> What happens if the job id referred in the `openqa_var_JOB_ID` is not found on OSD instance ? It could happens if an ID does not refer to a job on OSD but on some other instance.

The job _should_ have a known tag.

Reply:
> This is already the case within `webui`. PCW is not generic as this is our tool for cleanup.

I think we need to change this . see my comment above . we need to drop this dependency to **certain** openQA instance . Each set of tags should contain not only job id  but also server on which this job was running. 

> > What happens if the job id referred in the `openqa_var_JOB_ID` is not found on OSD instance ? It could happens if an ID does not refer to a job on OSD but on some other instance.
> 
> The job _should_ have a known tag.

I think Michele meant here is something what I said originally - we should cover the case that VM was created NOT by OSD.

Reply:
> I think we need to change this . see my comment above . we need to drop this dependency to **certain** openQA instance . 

Will see how feasible it is to implement this change in this PR>

> Each set of tags should contain not only job id but also server on which this job was running.

Fully agree, and we should not have non-URL tag values with `openqa-suse-de`.

> I think Michele meant here is something what I said originally - we should cover the case that VM was created NOT by OSD.

Yes.